modules/cli/src/main/java/org/apache/ignite/cli/core/repl/context/CommandLineContextProvider.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.cli.core.repl.context;
+
+import java.io.PrintWriter;
+import picocli.CommandLine;
+
+/**
+ * Provider of {@link CommandLineContext}.
+ */
+public class CommandLineContextProvider {
+
+    private static volatile CommandLine cmd;
```

Comment:
I wonder if having a static context can somehow mess up tests

Reply:
Yeah, I agree with you, but I specially leave it here as tech debt. I will add link to ticket for with this debt, are you against it?

Reply:
It's fine with me

Reply:
Ok, cool. Added link to issue